src/configure-balena.sh

```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC2209
+
+set -a
+
+DNS_TLD=${DNS_TLD:-$BALENA_TLD}
+ROOT_CA=${ROOT_CA:-$BALENA_ROOT_CA}
+BALENA_DEVICE_UUID=${BALENA_DEVICE_UUID:-balena}
+CONF=${CONF:-/balena/${BALENA_DEVICE_UUID}.${DNS_TLD}.env}
+CERTS=${CERTS:-/certs}
+
+# hostnames wthout BALENA_ prefix
+declare -A HOST_ENVVARS
```

Comment:
I know this isn't introduced by this PR, but the list of hostnames/services in here is an abomination.

Reply:
@dfunckt agree, but this one is a bit tricky and starts bleeding out of bash into something more structured. The `HOST_ENVVARS` and `SENTRY_ENVVARS` are just dictionaries, but `API_KEYS` also defines relationships between two or more env vars (e.g.):

```
API_KEYS[API_VPN_SERVICE_API_KEY]="${API_KEYS[API_SERVICE_API_KEY]}"
```

If we PR changes to both API and VPN to have these renamed to have the same name (and others), then `API_KEYS` becomes another simple key=value store. Though I am still not sure how we would populate the `HOST_ENVVARS` from another env var, aside from doing some nasty base64 encoding and expanding with `eval`.

Additionally, if confd had a mechanism to default env vars like in the shell, we could move a lot of logic out of the configure-balena script (e.g.)

```
MONITOR_OAUTH_COOKIE_SECRET=${MONITOR_OAUTH_COOKIE_SECRET:-$(openssl rand -hex 16)}
```

If it does already, I've not found it...  cc @Page-